File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int prev = 0, ans = 0;
        for (string s: bank) {
            int count = 0;
            for (int i = 0; i < s.length(); i++) {
                if (s[i] == '1') {
                    count++;
                }
            }
            if (count > 0) {
                ans += prev * count;
                prev = count;
            }
        }
        return ans;
    }
};

File: LeetCode/2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        prev , ans = 0 , 0
        for s in bank:
            count = 0
            for i in range(len(s)):
                if s[i] == '1':
                    count += 1
            if count > 0:
                ans += prev * count
                prev = count
        return ans